.github/workflows/ci.yml

```diff
@@ -302,9 +302,17 @@ jobs:
           username: ${{ secrets.DOCKER_USR }}
           password: ${{ secrets.DOCKER_PSW }}
 
-      - name: Publish Artifacts to S3, DockerHub
+      - name: Login to Upbound
+        uses: docker/login-action@v1
+        if: env.UPBOUND_MARKETPLACE_PUSH_ROBOT_USR != ''
+        with:
+          registry: xpkg.upbound.io
+          username: ${{ secrets.UPBOUND_MARKETPLACE_PUSH_ROBOT_USR }}
+          password: ${{ secrets.UPBOUND_MARKETPLACE_PUSH_ROBOT_PSW }}
+
+      - name: Publish Artifacts to S3, Marketplace, DockerHub
```

Comment:
Do we need to do any promoting (i.e. the next step in this job)?

Reply:
Since `xpkg.upbound.io/crossplane` is added to `REGISTRY_ORGS` it will be automatically included in the promotion stage as well. However, mirroring the check for the env var in the step could make sense for consistency.

Reply:
Added the check